pymc4/_model.py

```diff
@@ -83,8 +83,14 @@ def _evaluate(self):
     def make_log_prob_function(self):
         """Return the log probability of the model."""
 
-        def log_prob(*args):
-            context = contexts.InferenceContext(args, expected_vars=self._forward_context.vars)
+        def log_prob(**kwargs):
+            vars = self._forward_context.vars
+            if set(self._observations.keys()).intersection(set(kwargs.keys())):
+                raise ValueError('Passed variable already specified.')
+            kwargs.update(self._observations)
+            if len(kwargs) != len(vars):
```

Comment:
just realized we should probably check more than just len, otherwise len({'x','y'}) == len({'z','a'})

Reply:
Yes, related to Ravin's comment above. If we moved from a list to a dictionary, we could compare keys.

Reply:
Fixed in #118